pupil-api/src/services/azure-queue.service.ts

```diff
@@ -0,0 +1,27 @@
+import * as azure from 'azure-storage'
+import * as winston from 'winston'
+
+let azureQueueService // cache the queueService for repeated use
+
+export function addMessage (queueName: string, payload: object, queueService?: any) {
+  if (queueName === null || queueName.length < 1) {
+    throw new Error('Missing queueName')
+  }
+
+  if (!queueService) {
+    // If we have not been provided with a queueService, assume we want an Azure one.
+    if (!azureQueueService) {
+      azureQueueService = azure.createQueueService()
+    }
+    queueService = azureQueueService
+  }
+
+  const message = JSON.stringify(payload)
```

Comment:
@jon-shipley what happens if the payload is null|empty|undefined?

Reply:
I don't think this service needs to care.  In any case in the flow in the pupil-api the message is guaranteed:

```
    // Emit a successful login to the queue
    const pupilLoginMessage = {
      checkCode: result.checkCode._,
      loginAt: new Date(),
      version: 1
    }
    azureQueueService.addMessage('pupil-login', pupilLoginMessage)
```


And `JSON.stringify` will handle those cases you mentioned without throwing an error:

```
> JSON.stringify(null)
'null'
> JSON.stringify()
undefined
> JSON.stringify(undefined)
undefined
> 
```